packages/block-library/src/paragraph/edit.js

```diff
@@ -45,174 +45,215 @@ const applyFallbackStyles = withFallbackStyles( ( node, ownProps ) => {
 	};
 } );
 
-class ParagraphBlock extends Component {
-	constructor() {
-		super( ...arguments );
-
-		this.toggleDropCap = this.toggleDropCap.bind( this );
-	}
-
-	toggleDropCap() {
-		const { attributes, setAttributes } = this.props;
-		setAttributes( { dropCap: ! attributes.dropCap } );
-	}
-
-	getDropCapHelp( checked ) {
-		return checked ? __( 'Showing large initial letter.' ) : __( 'Toggle to show a large initial letter.' );
-	}
-
-	render() {
-		const {
-			attributes,
-			setAttributes,
-			mergeBlocks,
-			onReplace,
-			className,
-			backgroundColor,
-			textColor,
-			setBackgroundColor,
-			setTextColor,
-			fallbackBackgroundColor,
-			fallbackTextColor,
-			fallbackFontSize,
-			fontSize,
-			setFontSize,
-			isRTL,
-		} = this.props;
-
-		const {
-			align,
-			content,
-			dropCap,
-			placeholder,
+function useAttributeSetter( attribute, setAttributes ) {
+	return useCallback(
+		( newValue ) => {
+			setAttributes( { [ attribute ]: newValue } );
+		},
+		[ attribute, setAttributes ]
+	);
+}
+
+function ParagraphToolbar( { direction, setDirection } ) {
+	const isRTL = useSelect( ( select ) => {
+		return select( 'core/block-editor' ).getSettings().isRTL;
+	} );
+	const toolbarControls = useMemo(
```

Comment:
In the previous version, we were not using a memoized prop, do you think it's really necessary here?  I always wonder if we're over-optimizing sometimes, My personal opinion would be to start with a non-optimized version unless we measure change in performance.